### PR TITLE
Fix BeatmapSetOverlay not displaying Fails/Retries/Ratings

### LINKED ISSUE #4928: User Rating / Rating Spread and Points of failure does not show up in the Beatmap overlay
**Describe the bug:** 
Looking at Beatmaps in the client overlay does not show the rating / spread or the Points of Failure. 

**Screenshots or videos showing encountered issue:**
Web: 
image

Client:
image

**osu!lazer version:** 
2019.530.0
**Logs:** 
[network.log](https://github.com/ppy/osu/files/3256735/network.log)
[database.log](https://github.com/ppy/osu/files/3256753/database.log)
[runtime.log](https://github.com/ppy/osu/files/3256754/runtime.log)

### PR DESCRIPTION
**Pull request by smoogipoo**
Fixes #4928.

1. Metrics have been properly consolidated into `BeatmapSetInfo` and `BeatmapInfo`.
2. `APIBeatmapMetrics` and `GetBeatmapDetailsRequest` have been removed.
3. `BeatmapDetails` now performs a `GetBeatmapRequest`, and retrieves metrics from the appropriate set/beatmap structure.
4. Test scenes refactored/improved a bit, but previous functionality maintained for old tests.

### CODE REVIEW

**Comment by smoogipoo** [comment]
Didn't realise this was touching a lot of the same code as #4986, there'll be a few conflicting changes but hopefully it's contained enough.

**Comment by peppy** [comment]
@smoogipoo i've made some changes to `updateMetrics()` as it was previously a bit hard to follow data flow. now it uses the local reference to beatmap and doesn't pass it through.

i also moved the schedule logic as a safety precaution, and fixed a potential issue in the failure case (it wasn't checking whether the beatmap was still current before making changes), although it's worth nothing that that would have just triggered a second animation/update and not an incorrect display.

please double-check!

**Review by peppy — APPROVED** [review]

**Comment by smoogipoo** [comment]
Seems fine to me
